### PR TITLE
Add checks for schematron assert ids

### PR DESCRIPTION
**Pull request by bertrand-lorentz**

### CODE REVIEW

File: src/main/java/eu/europa/ted/eforms/sdk/analysis/util/SchematronParser.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  public static SchematronFile loadSchematronFile(Path schematronFilePath) {
+    logger.debug("Loading Schematron file " + schematronFilePath.toString());
+
+    IReadableResource schematron = new FileSystemResource(schematronFilePath);
+    // Resolve all included files, so that they also get loaded.
+    final IMicroDocument doc = SchematronHelper.getWithResolvedSchematronIncludes(schematron,
+        e -> handleError(e, schematronFilePath));
+
+    if (doc == null) {
+      logger.error("Schematron file %s could not be loaded as XML", schematronFilePath);
```
Comment (rouschr) [inline src/main/java/eu/europa/ted/eforms/sdk/analysis/util/SchematronParser.java:36]: I suppose the log messages should contain {} instead of %s

**Review by rouschr — CHANGES_REQUESTED** [review]
Looks nice, it is OK since you know this subject very well and there are unit tests.

File: src/main/java/eu/europa/ted/eforms/sdk/analysis/util/SchematronParser.java (hunk 1 above)
Comment (bertrand-lorentz) [reply src/main/java/eu/europa/ted/eforms/sdk/analysis/util/SchematronParser.java]: Indeed, I used the wrong syntax. Fixed in added commit.

**Review by rouschr — DISMISSED** [review]
As there are unit tests for valid and invalid case and you know Schematron well, the other parts should be OK.

**Comment by bertrand-lorentz** [comment]
Added a commit to fix the build with the latest EFX toolkit.
Not related to the other changes, but no reason to fix this separately.

**Review by rouschr — COMMENTED** [review]
OK for me

**Pull request merged**